pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -143,6 +148,11 @@ public HelixServerStarter(String helixClusterName, String zkAddress, PinotConfig
         new HelixConfigScopeBuilder(ConfigScopeProperty.PARTICIPANT, _helixClusterName).forParticipant(_instanceId)
             .build();
 
+    _environmentProperties = new HashMap<>();
+
+    // Fetch Environment Specific Configs
+    fetchEnvironmentSpecificConfigs();
```

Comment:
Change `fetchEnvironmentSpecificConfigs()` to return a map or `null`
```suggestion
    _environmentProperties = fetchEnvironmentProperties();
```